source/extensions/filters/http/jwt_authn/verifier.cc

```diff
@@ -299,22 +290,38 @@ class AnyVerifierImpl : public BaseGroupVerifierImpl {
     if (completion_state.is_completed_) {
       return;
     }
-    // For RequireAny: usually it returns the error from the last provider.
-    // But if a Jwt is not for a provider, its auth returns JwtMissed or JwtUnknownIssuer.
-    // Such error should not be used as the final error if there are other valid errors.
-    if (status != Status::Ok && status != Status::JwtMissed && status != Status::JwtUnknownIssuer) {
-      completion_state.any_valid_error_ = status;
-    }
-    if (++completion_state.number_completed_children_ == verifiers_.size() ||
-        Status::Ok == status) {
+
+    // If any of children is OK, return OK
+    if (Status::Ok == status) {
       completion_state.is_completed_ = true;
-      Status final_status = status;
-      if (status != Status::Ok && completion_state.any_valid_error_ != Status::Ok) {
-        final_status = completion_state.any_valid_error_;
+      completeWithStatus(status, context);
+      return;
+    }
+
+    // Then wait for all children to be done.
+    if (++completion_state.number_completed_children_ == verifiers_.size()) {
+      // Aggregate status from children, group into either Missed error or other failures
+      Status final_status = Status::JwtMissed;
+      for (const auto& it : verifiers_) {
+        Status child_status = context.getCompletionState(it.get()).status_;
+        if (child_status != Status::JwtMissed && child_status != Status::JwtUnknownIssuer) {
```

Comment:
nit: can you keep the old comment about the reason for specially handling JwtMissed and JwtUnknownIssuer? I find it is actually useful to understand the logic here.

Reply:
Update the comment.